pylint/pyreverse/diadefslib.py

```diff
@@ -67,6 +67,16 @@ def _get_levels(self) -> tuple[int, int]:
         """Help function for search levels."""
         return self.anc_level, self.association_level
 
+    def _should_include_by_depth(self, node: nodes.NodeNG) -> bool:
+        """Check if a node should be included based on depth."""
+        # If max_depth is not set, include all nodes
+        if self.config.max_depth is None:
+            return True
+
+        # For other nodes, calculate depth based on their root module
+        module_depth = node.root().name.count(".")
+        return module_depth <= self.config.max_depth  # type: ignore[no-any-return]
```

Comment:
```suggestion
        # If max_depth is not set, include all nodes, otherwise, calculate depth based on 
        # node's root module
        return self.config.max_depth is None or bool(node.root().name.count(".") <= self.config.max_depth)
```

The bool cast due to https://stackoverflow.com/a/75865051/2519059 (let me know what you think).